os/net/mac/tsch/tsch-queue.c

```diff
@@ -211,16 +213,14 @@ tsch_queue_remove_nbr(struct tsch_neighbor *n)
   if(n != NULL) {
     if(tsch_get_lock()) {
 
-      /* Remove neighbor from list */
-      list_remove(neighbor_list, n);
-
       tsch_release_lock();
 
       /* Flush queue */
       tsch_queue_flush_nbr_queue(n);
 
-      /* Free neighbor */
-      memb_free(&neighbor_memb, n);
+      /* Unlock and free neighbor */
+      nbr_table_unlock(tsch_neighbors, n);
```

Comment:
No need to unlock really, remove will do the job ("Locked" only means "not GC-able")

Reply:
Thanks, updated the code.